test/nettests/utils.cpp

```diff
@@ -0,0 +1,197 @@
+        REQUIRE(cannot_read == "./test/fixtures/urls.txt");
+    }
+
+    SECTION("When the randomize_input option is invalid") {
+        mk::ErrorOr<std::deque<std::string>> maybe_result =
+            mk::nettests::process_input_filepaths(
+                true, {"./test/fixtures/urls.txt"}, "IT",
+                {{"randomize_input", "antani"}}, mk::Logger::global(), nullptr,
+                nullptr);
+        REQUIRE(!maybe_result);
+        REQUIRE(maybe_result.as_error() == mk::ValueError());
+    }
+
+    SECTION("The randomize functionality is invoked when requested") {
+        REQUIRE_THROWS_AS(
+            (mk::nettests::process_input_filepaths_impl<
+                mk::nettests::open_file_, mk::nettests::readline_,
+                intercept_randomize>(true, {"./test/fixtures/urls.txt"}, "IT",
+                                     {{"randomize_input", "1"}},
+                                     mk::Logger::global(), nullptr, nullptr)),
+            mk::MockedError);
+    }
+
+    SECTION("The randomize functionality is invoked by default") {
+        REQUIRE_THROWS_AS((mk::nettests::process_input_filepaths_impl<
+                              mk::nettests::open_file_, mk::nettests::readline_,
+                              intercept_randomize>(
+                              true, {"./test/fixtures/urls.txt"}, "IT", {},
+                              mk::Logger::global(), nullptr, nullptr)),
+                          mk::MockedError);
+    }
+
+    SECTION("The randomize functionality is not invoked when not requested") {
+        /* No randomize, hence here we should not see exceptions: */
+        mk::nettests::process_input_filepaths_impl<mk::nettests::open_file_,
+                                                   mk::nettests::readline_,
+                                                   intercept_randomize>(
+            true, {"./test/fixtures/urls.txt"}, "IT",
+            {{"randomize_input", "0"}}, mk::Logger::global(), nullptr, nullptr);
+    }
+
+    SECTION("The randomize primitive works as expected") {
+
+        /*
+         * Same approach used in test/nettests/runnable.cpp: it is random
+         * if within a reasonable number of permutations we get back a
+         * vector different from the original one.
+         */
+
+        std::deque<std::string> expect{"torproject.org",
+                                       "ooni.nu",
+                                       "neubot.org",
+                                       "archive.org",
+                                       "creativecommons.org",
+                                       "cyber.law.harvard.edu",
+                                       "duckduckgo.com",
+                                       "netflix.com",
+                                       "nmap.org",
+                                       "www.emule.com"};
+
+        std::deque<std::string> copy = expect;
+        auto count = 0;
+        for (; count < 8; ++count) {
```

Comment:
I was about to say this was a small number, but actually P(x) = {same permutation x 8} = 1/10!^8 ~ 3,32579699816295e-53.
I guess that's quite unlikely.